cmd/cert-checker/main.go

```diff
@@ -170,10 +175,11 @@ func (c *certChecker) checkCert(cert core.Certificate) (problems []string) {
 		if parsedCert.IsCA {
 			problems = append(problems, "Certificate can sign other certificates")
 		}
-		// Check the cert has the correct validity period +/- an hour
-		if parsedCert.NotAfter.Sub(cert.Issued) > checkPeriod+time.Hour {
+		// Check the cert has the correct validity period
+		expiryPeriod := parsedCert.NotAfter.Sub(parsedCert.NotBefore)
+		if expiryPeriod > checkPeriod {
 			problems = append(problems, "Certificate has a validity period longer than 90 days")
-		} else if parsedCert.NotAfter.Sub(cert.Issued) < checkPeriod-time.Hour {
+		} else if expiryPeriod < checkPeriod {
```

Comment:
If we give out a 90 cert (as folks say we will), won't this be try even on the exact day it's given out?